### PR TITLE
ENDOC-578 Update the RBAC tutorial for Entando 7.1

### PR DESCRIPTION
**Pull request by nshaw**

### CODE REVIEW

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 1]
```diff
@@ -164,32 +127,39 @@ Promote the admin user to a full `conference-admin` to reinstate the ability to
...
 
-In Kubernetes, Entando will automatically create client roles per the bundle plugin definition (see the [plugin definition](../../../docs/curate/bundle-details.md) for more information). These roles are created for the client specific to the microservice, e.g. `<docker username>-conference-server`. The client name is injected as an environment variable into the plugin container, so the annotations noted above will work in both local and Kubernetes environments.
+In Kubernetes, Entando will automatically create client roles per the bundle plugin definition (see the [plugin definition](../../../docs/curate/bundle-details.md) for more information). These roles are created for the client specific to the microservice, e.g. `pn-YOUR-SERVICE-ID-conference-ms`. The client name is injected as an environment variable into the plugin container, so the annotations noted above will work in both local and Kubernetes environments.
 
-#### Modify Security Checks for Kubernetes
+#### Keycloak Client options in an Entando Application
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:152]: Option (capitalized to match style of other headers)

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 2]
```diff
@@ -130,9 +93,9 @@ The `conference-admin` role should grant a user permission to delete Conferences
 
 To verify that a user without the `conference-admin` role is unable to call the delete API:
 
-1. Restart the microservice. By default this includes rebuilding any changed source files.
-2. Once the microservice is available, return to the MFE and try deleting one of the Conferences in the list 
-3. Verify that attempting to delete via the UI generates a `403 error` in the browser console and an error in the service logs similar to the following:
+1. Restart the microservice. By default, this includes rebuilding any changed source files.
+2. Return to the MFE and try deleting one of the Conferences in the list 
+3. Verify that attempting to delete a conference via the UI generates a `403 error` in the browser console. There should be an error in the service logs similar to the following:
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:98]: "conference" is capitalized in the step above, which is contextually similar

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 3]
```diff
@@ -30,80 +30,43 @@ The list of Conferences must be visible to only the `conference-user` and `confe
...
+ent bundle run conference-table
 ```
-4. When prompted to select a widget to run, choose the option corresponding to the tableWidget, e.g. `ui/widgets/conference/tableWidget`
 
-### Step 3: Access the tableWidget MFE
+### Step 3: Access the conference-table MFE
 
-1. In your browser, go to <http://localhost:3000>. This is typically the location of the tableWidget MFE.
-2. Access the tableWidget MFE with the default credentials of `username: admin`, `password: admin` 
+1. In your browser, go to <http://localhost:3000>.
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:53]: minor: remove period for consistency

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 4]
```diff
@@ -164,32 +127,39 @@ Promote the admin user to a full `conference-admin` to reinstate the ability to
 5. Confirm the delete icon is visible 
 6. Confirm a Conference can be successfully deleted from the list
 
-## Notes
-### Realm Roles versus Client Authorities
-This tutorial utilizes authorities. In Keycloak, authorities are roles mapped to a user for a specific client. It is possible to assign higher-level Realm Roles directly to users, e.g. `ROLE_ADMIN`, but this can result in collisions between applications using the same roles.
+### Step 10. Configure the roles in `entando.json`
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:130]: prefer this reference to `entando.json` which is not prefaced by "the"; minor but technically correct to *not* use "the"

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 5]
```diff
@@ -164,32 +127,39 @@ Promote the admin user to a full `conference-admin` to reinstate the ability to
...
+Entando can automatically add roles to your client (see [the notes below](#notes) for different client options) when your microservice is deployed.
 
-To implement Realm-assigned roles, the code above must be modified:
-- In the backend, use the annotation `@Secured('ROLE_ADMIN)` or `@PreAuthorize(hasRole('ROLE_ADMIN'))`
-- In the frontend, use `keycloak.hasRealmRole` instead of `keycloak.hasResourceRole` 
+1. Modify the `entando.json` by adding the following line to the `microservices/conference-ms`:
```
Comment (Lyd1aCla1r3) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:133]: remove "the" :)

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md [hunk 6]
```diff
@@ -30,80 +30,43 @@ The list of Conferences must be visible to only the `conference-user` and `confe
...
+### Step 3: Access the conference-table MFE
 
-1. In your browser, go to <http://localhost:3000>. This is typically the location of the tableWidget MFE.
-2. Access the tableWidget MFE with the default credentials of `username: admin`, `password: admin` 
+1. In your browser, go to <http://localhost:3000>.
+2. Access the conference-table MFE with the default credentials of `username: admin`, `password: admin` 
 
-> Note: Once authenticated, the message "No conferences are available" is generated. If you check your browser
-console, you should see a `403 (Forbidden)` error for the request made to `localhost:8080/services/conference/api/conferences`. This is expected because the admin user has not yet been granted the new role. 
+> Note: Once authenticated, the message "No conferences are available" is generated. If you check your browser console, you should see a `403 (Forbidden)` error for the request made to `localhost:8080/services/conference/api/conferences`. This is expected because the admin user has not yet been granted the new role.
```
Comment (jyunmitch) [inline vuepress/docs/next/tutorials/create/ms/add-access-controls.md:56]: backtick admin in last line

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md (hunk 5 above)
Comment (jyunmitch) [reply vuepress/docs/next/tutorials/create/ms/add-access-controls.md:133]: think it's correct either way, with or without the

**Review by jyunmitch — APPROVED** [review]

File: vuepress/docs/next/tutorials/create/ms/add-access-controls.md (hunk 5 above)
Comment (Lyd1aCla1r3) [reply vuepress/docs/next/tutorials/create/ms/add-access-controls.md:133]: it's not. https://english.stackexchange.com/questions/399258/should-i-use-the-before-file-names

**Pull request merged**